symfony/client/src/orangehrmAdminPlugin/pages/payGrade/EditPayGrade.vue

```diff
@@ -0,0 +1,153 @@
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+ -->
+
+<template>
+  <div class="orangehrm-background-container">
+    <div class="orangehrm-card-container">
+      <oxd-text tag="h6" class="orangehrm-main-title">
+        Edit Pay Grade
+      </oxd-text>
+
+      <oxd-divider />
+
+      <oxd-form :loading="isLoading" @submitValid="onSave">
+        <oxd-form-row>
+          <oxd-input-field
+            label="Name"
+            v-model="grade.name"
+            :rules="rules.name"
+            required
+          />
+        </oxd-form-row>
+
+        <oxd-divider />
+
+        <oxd-form-actions>
+          <required-text />
+          <oxd-button
+            type="button"
+            displayType="ghost"
+            label="Cancel"
+            @click="onCancel"
+          />
+          <submit-button />
+        </oxd-form-actions>
+      </oxd-form>
+    </div>
+  </div>
+  <pay-grade-currency :pay-grade-id="payGradeId"></pay-grade-currency>
+</template>
+
+<script>
+import {navigate, reloadPage} from '@orangehrm/core/util/helper/navigation';
+import {APIService} from '@/core/util/services/api.service';
+import {
+  required,
+  shouldNotExceedCharLength,
+} from '@orangehrm/core/util/validation/rules';
+import PayGradeCurrency from '@/orangehrmAdminPlugin/pages/payGrade/PayGradeCurrency.vue';
+
+export default {
+  components: {
+    'pay-grade-currency': PayGradeCurrency,
+  },
+
+  props: {
+    payGradeId: {
+      type: String,
+      required: true,
+    },
+  },
+
+  setup() {
+    const http = new APIService(
+      window.appGlobal.baseUrl,
+      'api/v2/admin/pay-grades',
+    );
+    return {
+      http,
+    };
+  },
+
+  data() {
+    return {
+      isLoading: false,
+      grade: {
+        id: '',
+        name: '',
+      },
+      rules: {
+        name: [required, shouldNotExceedCharLength(50)],
+      },
+      errors: [],
+    };
+  },
+
+  methods: {
+    onSave() {
+      this.isLoading = true;
+      this.http
+        .update(this.payGradeId, {
+          name: this.grade.name,
+        })
+        .then(() => {
+          return this.$toast.updateSuccess();
+        })
+        .then(() => {
+          reloadPage();
+        });
+    },
+    onCancel() {
+      navigate('/admin/viewPayGrades');
+    },
+  },
+  created() {
+    this.isLoading = true;
+    this.http
+      .get(this.payGradeId)
+      .then(response => {
+        const {data} = response.data;
+        this.grade.id = data.id;
+        this.grade.name = data.name;
+        return this.http.getAll();
+      })
+      .then(response => {
+        const {data} = response.data;
+        this.rules.name.push(required);
```

Comment:
since these rules are defined in `data()` method. no need to add it again,

Reply:
Done